### PR TITLE
[GOBBLIN-1508] update some cluster side metrics from guage to meter

### PR DESCRIPTION
**Pull request by arjun4084346**
add some new metrics

Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!
@jack-moseley  @Will-Lo  please review

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1508

### Description
- [x] Here are some details about my PR, including screenshots (if applicable):
add a few metrics in gobblin cluster and update some from counter to meter, because meter can give both count and rate of an entity

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
trivial changes

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by arjun4084346** [comment]
@aplex  please review

File: gobblin-modules/gobblin-service-kafka/src/main/java/org/apache/gobblin/service/StreamingKafkaSpecConsumer.java [hunk 1]
```diff
@@ -196,44 +197,48 @@ public void onCancelJob(URI cancelledJobURI) {
...
-          StreamingKafkaSpecConsumer.this._jobMonitor.getRemovedSpecs().getCount() : 0;
+      return StreamingKafkaSpecConsumer.this._jobMonitor.getCancelledSpecs() != null ?
+          StreamingKafkaSpecConsumer.this._jobMonitor.getCancelledSpecs().getCount() : 0;
     }
 
     private long getMessageParseFailures() {
-      return StreamingKafkaSpecConsumer.this._jobMonitor.getMessageParseFailures() != null?
+      return StreamingKafkaSpecConsumer.this._jobMonitor.getMessageParseFailures() != null ?
           StreamingKafkaSpecConsumer.this._jobMonitor.getMessageParseFailures().getCount():0;
     }
```
Comment (jack-moseley) [inline gobblin-modules/gobblin-service-kafka/src/main/java/org/apache/gobblin/service/StreamingKafkaSpecConsumer.java:243]: Are these methods not used anymore? Can we just remove them if so?

Comment (jack-moseley) [reply gobblin-modules/gobblin-service-kafka/src/main/java/org/apache/gobblin/service/StreamingKafkaSpecConsumer.java:243]: How are they being used? When I look at usages of this `private long getNewSpecs()` method I only see it used by the constructor right above (line 221), and you are removing that usage right?

Comment (arjun4084346) [reply gobblin-modules/gobblin-service-kafka/src/main/java/org/apache/gobblin/service/StreamingKafkaSpecConsumer.java]: Ah, okay. I messed two PRs. removed this code

**Review by jack-moseley — APPROVED** [review]

**Review by aplex — APPROVED** [review]

**Pull request merged**